addons/web/static/src/views/fields/properties/property_value.js

```diff
@@ -77,9 +77,11 @@ export class PropertyValue extends Component {
     get propertyValue() {
         const value = this.props.value;
 
-        if (this.props.type === "float") {
+        if (this.props.type === "integer") {
+            return value || 0;
+        } else if (this.props.type === "float") {
             // force to show at least 1 digit, even for integers
-            return value;
+            return value || 0.0;
```

Comment:
I believe you can do something like this because 0.0 and 0 are just equal.
```js
if (["float", "integer"].includes(this.props.type)) {
  return value || 0;
}
```